pkg/vm/vm.go

```diff
@@ -1602,8 +1602,32 @@ func (v *VM) handleException() {
 		}
 		pop++
 		ictxv = ictxv.Next()
+		if ictxv == nil {
+			break
+		}
 		ictx = ictxv.Value().(*Context)
 	}
+	throwUnhandledException(v.uncaughtException)
+}
+
+// throwUnhandledException gets exception message from the provided stackitem and panics.
+func throwUnhandledException(item stackitem.Item) {
+	msg := "An unhandled exception was thrown"
```

Comment:
"unhandled exception" is enough, that's where it's OK to have some difference to C#.